qiita_db/analysis.py

```diff
@@ -947,6 +945,11 @@ def _build_biom_tables(self, grouped_samples, rarefaction_depth=None,
                     else:
                         new_table = new_table.merge(biom_table)
 
+                if not new_table or len(new_table.ids()) == 0:
+                        raise RuntimeError(
+                            "All samples filtered out from Artifact %s due "
```

Comment:
This text is not clear. "All sample filtered from artifact %s: No samples in artifact matched selected samples."